Gems/Atom/RHI/Code/Include/Atom/RHI/MultiDeviceDrawPacketBuilder.h

```diff
@@ -0,0 +1,128 @@
+            DrawPacketBuilder::DrawRequest GetDeviceDrawRequest(int deviceIndex);
+
+            //! The filter tag used to direct the draw item.
+            DrawListTag m_listTag;
+
+            //! The stencil ref value used for this draw item.
+            uint8_t m_stencilRef{};
+
+            //! The array of stream buffers to bind for this draw item.
+            AZStd::span<const MultiDeviceStreamBufferView> m_streamBufferViews;
+
+            //! Shader resource group unique for this draw request
+            const MultiDeviceShaderResourceGroup* m_uniqueShaderResourceGroup{};
+
+            //! The pipeline state assigned to this draw item.
+            const MultiDevicePipelineState* m_pipelineState{};
+
+            //! The sort key assigned to this draw item.
+            DrawItemSortKey m_sortKey{};
+
+            //! Mask for filtering the draw item into specific render pipelines.
+            //! We use a mask because the same item could be reused in multiple pipelines. For example, a simple
+            //! depth pre-pass could be present in multiple pipelines.
+            DrawFilterMask m_drawFilterMask = DrawFilterMaskDefaultValue;
+
+            //! A map of all device-specific StreamBufferViews, indexed by the device index
+            //! This additional cache is needed since device-specific StreamBufferViews are returned as objects
+            //! and the device-specific DrawItem holds a pointer to it.
+            AZStd::unordered_map<int, AZStd::vector<StreamBufferView>> m_deviceStreamBufferViews;
+        };
+
+        explicit MultiDeviceDrawPacketBuilder(RHI::MultiDevice::DeviceMask deviceMask)
+            : m_deviceMask{ deviceMask }
+        {
+            auto deviceCount{ RHI::RHISystemInterface::Get()->GetDeviceCount() };
+
+            for (int deviceIndex = 0; deviceIndex < deviceCount; ++deviceIndex)
+            {
+                if ((AZStd::to_underlying(m_deviceMask) >> deviceIndex) & 1)
+                {
+                    m_deviceDrawPacketBuilders.emplace(deviceIndex, DrawPacketBuilder());
+                }
+            }
+        }
+
+        // NOTE: This is configurable; just used to control the amount of memory held by the builder.
+        static const size_t DrawItemCountMax = 16;
+
+        void Begin(IAllocator* allocator);
+
+        void SetDrawArguments(const MultiDeviceDrawArguments& drawArguments);
+
+        void SetIndexBufferView(const MultiDeviceIndexBufferView& indexBufferView);
+
+        void SetRootConstants(AZStd::span<const uint8_t> rootConstants);
+
+        void SetScissors(AZStd::span<const Scissor> scissors);
+
+        void SetScissor(const Scissor& scissor);
+
+        void SetViewports(AZStd::span<const Viewport> viewports);
+
+        void SetViewport(const Viewport& viewport);
+
+        void AddShaderResourceGroup(const MultiDeviceShaderResourceGroup* shaderResourceGroup);
+
+        void AddDrawItem(MultiDeviceDrawRequest& request);
```

Comment:
Add a quick comment per method in order to have better documentation for new code. Thanks

Reply:
done!